### PR TITLE
Change field tag by entity

### PR DESCRIPTION
**Pull request by Steackachais**
For futur functionnality, i split the field in Json Tags and i add a entity Tag, 
Use command 
- php bin/console doctrine:migrations:migrate

And

-  php bin/console app:update:CreateRelationJobTag

You can see and compare result in the route /entityTag
I push in the futur a other version with the field Tag delete and the provider use this entity.

### CODE REVIEW

File: migrations/Version20221109081919.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20221109081919 extends AbstractMigration
```
Comment (qdequippe) [inline migrations/Version20221109081919.php:13]: You can remove it it's empty

File: migrations/Version20230109104110.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('DROP TABLE source');
```
Comment (qdequippe) [inline migrations/Version20230109104110.php:23]: Delete this one also source table should exists

File: migrations/Version20230109104306.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('DROP TABLE source');
```
Comment (qdequippe) [inline migrations/Version20230109104306.php:23]: duplicate

File: src/Command/UpdateTagCommand.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        private readonly TagsRepository $tagsRepository,
+        private readonly JobRepository $jobRepository
+    ) {
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $oJobs = $this->jobRepository->findAll();
```
Comment (qdequippe) [inline src/Command/UpdateTagCommand.php:36]: There is a lot of jobs, you should select only a subset (last 100?) or batch it

File: src/Controller/JobController.php [hunk 5]
```diff
@@ -59,6 +59,14 @@ public function rss(): Response
         ]);
     }
 
+    #[Route('/entityTag', name: 'job_index_entity_tag', defaults: ['_format' => 'html'], methods: ['GET']), ]
```
Comment (qdequippe) [inline src/Controller/JobController.php:62]: What is the purpose of this route?

File: src/Entity/Job.php [hunk 6]
```diff
@@ -133,6 +135,9 @@ class Job
     #[ApiFilter(filterClass: SearchFilter::class, strategy: SearchFilterInterface::STRATEGY_PARTIAL)]
     private ?LocationType $locationType = null;
 
+    #[ORM\ManyToMany(targetEntity: Tags::class, mappedBy: 'job')]
+    private Collection $tag;
```
Comment (qdequippe) [inline src/Entity/Job.php:139]: `$tags` maybe?

File: src/Entity/Tags.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+namespace App\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use ApiPlatform\Metadata\ApiResource;
+
+#[ORM\Entity]
+#[ApiResource]
+class Tags
```
Comment (qdequippe) [inline src/Entity/Tags.php:12]: It's a single tag or multiple tags ?

File: src/Entity/Tags.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\ManyToMany(targetEntity: Job::class, inversedBy: 'tag')]
+    private Collection $job;
```
Comment (qdequippe) [inline src/Entity/Tags.php:20]: Do you need this relation?

File: src/Entity/Tags.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+use ApiPlatform\Metadata\ApiResource;
+
+#[ORM\Entity]
+#[ApiResource]
+class Tags
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
```
Comment (qdequippe) [inline src/Entity/Tags.php:17]: Could you use Uuid instead as done in all entities?

File: src/Repository/TagsRepository.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    /**
+     * @param $label
+     * @throws NonUniqueResultException
+     */
+    public function findOneByLabelCI(string $label): ?Tags
```
Comment (qdequippe) [inline src/Repository/TagsRepository.php:47]: What "CI" means?

File: templates/job/_job_card.html.twig [hunk 11]
```diff
@@ -29,16 +29,27 @@
                     <li class="d-flex align-items-center gap-1">
                         <i icon-name="map-pin" class="fs-6"></i>
                         {{ job.location }}
-                        {% if job.locationType %}
```
Comment (qdequippe) [inline templates/job/_job_card.html.twig:32]: Why this change?

File: src/Controller/JobController.php (hunk 5 above)
Comment (Steackachais) [reply src/Controller/JobController.php:62]: It's a temporary route for check the display the entity tag.

File: src/Entity/Tags.php (hunk 8 above)
Comment (Steackachais) [reply src/Entity/Tags.php:20]: Yes , a job have many Tag and a Tag have many job? I don't understand this question

File: src/Entity/Tags.php (hunk 7 above)
Comment (Steackachais) [reply src/Entity/Tags.php:12]: It's a single Tag but for the update i can't remove/change thhe name to fiield Tag

Comment (Steackachais) [reply src/Entity/Tags.php:12]: A single Tag, it's a error . i need to change that.

File: src/Entity/Job.php (hunk 6 above)
Comment (Steackachais) [reply src/Entity/Job.php:139]: Yes but there can't remove/change the past fielsName tag without delete data.

File: src/Entity/Tags.php (hunk 9 above)
Comment (Steackachais) [reply src/Entity/Tags.php:17]: Yes i can but i don't like use type string for a ID

File: src/Entity/Tags.php (hunk 8 above)
Comment (qdequippe) [reply src/Entity/Tags.php]: Ok but why bi directional?

File: src/Entity/Tags.php (hunk 9 above)
Comment (qdequippe) [reply src/Entity/Tags.php]: > Yes i can but i don't like use type string for a ID

Ok, why? :)

File: src/Repository/TagsRepository.php (hunk 10 above)
Comment (qdequippe) [reply src/Repository/TagsRepository.php]: Why do you have to name function like this? Is "case insensitive" is not always required?

File: src/Entity/Tags.php (hunk 9 above)
Comment (Steackachais) [reply src/Entity/Tags.php]: Because I learn and I read this.  It's more optimized for "join" and needs less space. UUID.It's good for security for user's interface. You can use both .

Comment (qdequippe) [reply src/Entity/Tags.php]: > optimized for "join"

We don't have a lot of data here so it's not a problem and we don't have a lot of relations between objects

> needs less space

Even is UUID is stored as binary?

> user's interface

UUID for user interface?

Comment (Steackachais) [reply src/Entity/Tags.php]: > We don't have a lot of data here so it's not a problem and we don't have a lot of relations between objects
 Well so do nothing. After executing the command, I have ~2000 tuples, I agree, it's not a lot. 
> Even if UUID is stored as binary?
It seems to me that it's not the most important for me.
> UUID for user interface?
UUID use in href better than an ID for example

File: src/Repository/TagsRepository.php (hunk 10 above)
Comment (Steackachais) [reply src/Repository/TagsRepository.php]: > Why do you have to name function like this?
I prefer to specify to avoid surprises.

> Is "case insensitive" is not always required?
If only, Win versus Linux
